mobbex/Models/Transaction.php

```diff
@@ -68,45 +69,122 @@ class Transaction extends AbstractModel
+            'limit'     => "LIMIT $limit",
+        ];
+        // Make request to db to get an array of the transaction
+        $result = \Db::getInstance()->executeS(
+            $query['operation'] . ' FROM ' . $query['table'] . ' ' . $query['condition'] . ' ' . $query['order'] . ' ' . $query['limit']
+        );
 
-        $trx->save();
+        if ($limit <= 1)
+            return isset($result[0]) ? $result[0] : null;
 
-        return $trx;
+        return !empty($result) ? $result : null;
     }
-
+    
     /**
-     * Get the transactions from the db and returns an array of \Mobbex\PS\Checkout\Models\Transaction objects.
-     * If param $parent is true, it returns only the parent webhook
+     * Creates sql 'WHERE' statement with an associative array.
      * 
-     * @param int $cart_id
-     * @param bool $parent 
+     * @param array $conditions
+     * 
+     * @return string $condition
      * 
-     * @return array|object
      */
-    public static function getTransactions($cart_id, $parent = false)
+    public static function getCondition($conditions)
     {
-        $data = \Db::getInstance()->executes('SELECT * FROM ' . _DB_PREFIX_ . 'mobbex_transaction' . ' WHERE cart_id = ' . $cart_id . ($parent ? ' and parent = 1' : '') . ' ORDER BY id DESC');
+        $i = 0;
+        $condition = '';
 
-        if (!$parent) {
-            $childs = [];
-            foreach ($data as $value)
-                $childs[] = new \Mobbex\PS\Checkout\Models\Transaction($value['id']);
-            return $childs;
+        foreach ($conditions as $key => $value) {
+            if($i < 1)
+                $condition .= "WHERE `$key`='{$value}' ";
+            else
+                $condition .= " AND `$key`='$value' ";
+            $i++;
         }
 
-        return !empty($data[0]) ? new \Mobbex\PS\Checkout\Models\Transaction($data[0]['id']) : new \Mobbex\PS\Checkout\Models\Transaction();
+       return $condition;
+    }
+
+    /**
+     * Get childs data from transaction or from webhook and create an array of child transactions(type object)
+     * 
+     * @return array $childs (funciona como nuevo atributo que guarda childs trx)
+     * 
+     */
+    public function getChilds()
+    {
+        if (!empty($this->childs)){
+            // Create a transaction instance and set it with formated webhook childs data as an array
+            foreach (json_decode($this->childs, true) as $child)
+                $childsData[] = (new \Mobbex\PS\Checkout\Models\Transaction)->loadFromWebhookData($child);
+        } else {
+            // Takes all the child transactions of the table, instantiates them, and stores them in an array (support for the old way webhooks arrived)
+            $childs = self::getData(['cart_id' => $this->cart_id, 'parent' => false], 0);
+            foreach ($childs as $childData)
+                $childsData[] = new Transaction($childData['id']);
+            }
+        $this->childs = $childsData;
```

Comment:
Ojo aca tendria que setearse con $childData['data'] ya que this->childs guarda la data cruda del webhook

Reply:
Dale! venia encodeado en json?
osea deberia ser `foreach ($childs as json_decode($childData['data']))` ?

Reply:
No, lo que estoy diciendo es que donde haces  $this->childs = $childsData; tendrias que hacer  $this->childs = $childsData['data']; ya que si no estas guardando en childs la data capada en lugar del webhook completo, y en childs lo que se guarda es la data completa, otro detalle a tener en cuenta es que $this->childs es un array de arrays, asi que además seria $this->childs[] = $childsData['data'];

recordá que cuando obtenes de base de datos, lo que obtenes es un arrray asociativo con una posicion por cada clumna con la data asignada a cada una, en la columna "data" de transactions guardamos la data cruda como llega del webhook